src/shared/ptyfwd.h

```diff
@@ -23,7 +23,7 @@ typedef enum PTYForwardFlags {
         PTY_FORWARD_DUMB_TERMINAL          = 1 << 3,
 } PTYForwardFlags;
 
-typedef int (*PTYForwardHandler)(PTYForward *f, int rcode, void *userdata);
+typedef int (*PTYForwardHandler)(sd_event *e, PTYForward *f, int rcode, void *userdata);
```

Comment:
still not a fan of adding this here. why not simply pass the context via the userdata pointer? it's not used so far in your tool, and should be just enough...